### PR TITLE
Added rev2 & move rev1+rev2 to parent folder

### PR DESCRIPTION
**Pull request by Aplyard**
Moved aplx6 to parent folder, renamed it to rev1 and added rev2

rev2 has different keymap.c from rev1

## Description

Aplx6 has a new rev and a new pcb with encoder & oled display support.
Made a parent folder named aplx, renamed aplx6 to rev and added rev2 folder and files.
Edited old rev1 readme.md to correct the make: example.
Sorry for any bad coding in advance :P
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).Screenshot 2020-11-18 at 8 27 44 PM

### CODE REVIEW

File: keyboards/aplx/aplx6/rev1/config.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #define DEVICE_VER      0x0001
 #define MANUFACTURER    Aplyard
 #define PRODUCT         Aplx6
-#define DESCRIPTION     Aplx6 MediaPad
+#define DESCRIPTION     Aplx6 MediaPad rev1
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev1/config.h:28]:
```suggestion
```

File: keyboards/aplx/aplx6/rev2/config.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xE0E0
+#define PRODUCT_ID      0x0040
+#define DEVICE_VER 		0x0002
+#define MANUFACTURER    Aplyard
+#define PRODUCT         Aplx6
+#define DESCRIPTION     Aplx6 Mediapad rev2
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/config.h:28]:
```suggestion
```

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,126 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 63, 35,231, 47,255, 62, 62, 61, 61, 61, 61, 61, 61, 62, 62, 63,239, 39,227, 63,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 15, 15,  8,  8,  8,  8,  0,  7, 15,  8,  8,  4, 15,  0,  0, 64, 65,111, 60, 15,  1,  0,  0,  3,  7, 12,  8,  8,  8,  0,  0,  0, 15, 15,  0,  0,  0,  8, 12,  0,  0,  0,  0,  0,  0,  0,  0,  8,  8,  8, 12,  6,  3,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 
+        0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  3,  2,  3,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  3,  2,  3,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,
+    };
+    //Switch between logos
+    switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c:85]:
```suggestion
    switch (get_highest_layer(layer_state)) {
```

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    }  
+}
+
+void oled_task_user(void) {  
+    render_logo();
+}
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c:103]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/aplx/aplx6/rev2/rev2.c [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#include "rev2.h"
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/rev2.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/aplx/aplx6/rev2/rev2.c [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+void matrix_init_kb(void) {
+	matrix_init_user();
+}
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/rev2.c:4]:
```suggestion
```

File: keyboards/aplx/aplx6/rev2/rev2.h [hunk 8]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/rev2.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/aplx/aplx6/rev2/rules.mk [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/aplx/aplx6/rev2/rules.mk:11]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c (hunk 3 above)
Comment (Aplyard) [reply keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c:16]: done it!

File: keyboards/aplx/aplx6/rev2/rev2.c (hunk 6 above)
Comment (Aplyard) [reply keyboards/aplx/aplx6/rev2/rev2.c:16]: done it!

File: keyboards/aplx/aplx6/rev2/rev2.h (hunk 8 above)
Comment (Aplyard) [reply keyboards/aplx/aplx6/rev2/rev2.h:16]: done it! thnx for the suggestions

File: keyboards/aplx/aplx6/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (noroadsleft) [inline keyboards/aplx/aplx6/rev2/rules.mk:24]: Some mild cleanup; we don't include the MIDI or Faux Clicky rules here anymore, unless they are enabled.

```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/aplx/aplx6/rev2/rules.mk [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+UNICODE_ENABLE = yes        # Unicode
+OLED_DRIVER_ENABLE = yes    # Enable Support for Oled Display
+ENCODER_ENABLE = yes		# Enable Support for Encoder
```
Comment (noroadsleft) [inline keyboards/aplx/aplx6/rev2/rules.mk:27]: tabs -> spaces

```suggestion
ENCODER_ENABLE = yes        # Enable Support for Encoder
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Also, more of a general question, but why not `aplyard` as the parent directory? You're using that as your Manufacturer name for the USB strings.

It's not wrong or anything, but it strikes me as unusual.

**Comment by noroadsleft** [comment]
Also, as this moves the rev1 in a way that's not backwards compatible with the previous workflow (the `make` command changes), this should target the `develop` branch.

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+	*  |   APLX6   |------|------|------| |
+	*  '----------------------------------'
+	*/
+  [_POWER] = LAYOUT(
+    KC_CALC, KC_MY_COMPUTER, TO(0),
+    KC_COPY, KC_CUT, KC_PASTE
+  )
+};
+
+static void render_logo(void) {
```
[the remark this answers is not in the document]
Comment (Aplyard) [reply keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c:78]: You are absolutely correct, and i was including the if statement when building it. My idea was that the OLED is mandatory thats why. Sure i will include it

File: keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */
+#include QMK_KEYBOARD_H
+
+#define _MEDIA 0
+#define _DOC 1
+#define _POWER 2
```
[the remark this answers is not in the document]
Comment (Aplyard) [reply keyboards/aplx/aplx6/rev2/keymaps/default/keymap.c]: Nice one! Thanks a ton!

**Comment by Aplyard** [comment]
> Also, more of a general question, but why not `aplyard` as the parent directory? You're using that as your Manufacturer name for the USB strings.
> 
> It's not wrong or anything, but it strikes me as unusual.

Hey and thanks for a the suggestions. I simply use Aplx as a Category name for anything that has to do with a Keyboard. At first i tried it as " /aplyard/aplx6/rev2/... " and with the " qmk compile -kb aplyard/aplx6/rev2  -km default " i was getting an error which i couldnt figure out. Simply by changing the parent folder to aplx, it was compiling ok.
Now after your suggestion i renamed the parent to "aplyard", i tried "cd" to qmk folder and did a 
" make aplyard/aplx6/rev2:default "  and it worked, don't know why. I guess i should have tried the make command from day1.

**Comment by Aplyard** [comment]
hello, is there anything else you want me to do? made all the suggested changes.
Thank you again

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**